### PR TITLE
Spr1 735

### PR DESCRIPTION
**Pull request by cchristelis**
**Intro**

This PR relates to the issue https://skaafrica.atlassian.net/browse/SPR1-735 
The concept of time on target was added to the metadata as "IntegrationTime"
The functional code was borrowed from another repository. (No good common place for this code was identified, ideally we would rather want to share this functionality than copy it across.)

**Tests**

pytest tests added. 
I'm not quite sure whether the setup.py was updated correctly.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -26,5 +26,12 @@
     platforms=["OS Independent"],
     keywords="meerkat ska",
     zip_safe=False,
-    use_katversion=True
+    use_katversion=True,
+    test_suite="tests",
+    setup_requires=[
+        "pytest-runner",
```
Comment (bmerry) [inline setup.py:32]: pytest-runner is [deprecated](https://pypi.org/project/pytest-runner/).

**Comment by bmerry** [comment]
> I'm not quite sure whether the setup.py was updated correctly.

To get Jenkins to run the tests, you also need:
- a test-requirements.txt, which contains all packages (including recursive dependencies) that need to be installed on top of those in requirements.txt for testing, and pinned where katsdpdockerbase's base requirements doesn't have default versions.
- a pytest.ini file (so that Jenkins knows to run pytest and not nosetests). It can be empty, but also take a look at something like katsdpmodels.

**Review by lmzxqtom — APPROVED** [review]
Some changes requested from Bruce, otherwise looks good to me. Thanks @cchristelis.

**Comment by cchristelis** [comment]
Thanks for the input @bmerry. I hope this is now correct.

**Comment by bmerry** [comment]
> Thanks for the input @bmerry. I hope this is now correct.

Are you able to push your changes to a branch within ska-sa/katsdpdata, rather than your fork? The Jenkins server doesn't build from forks (for security reasons), which makes it hard to tell.

**Comment by cchristelis** [comment]
@bmerry , there is now a second PR with these changes based off a branch in this repo: https://github.com/ska-sa/katsdpdata/pull/175

**Comment by ludwigschwardt** [comment]
Question: is "IntegrationTime" the appropriate key? That's taken from the "Images" section of the schema. What if there are no images associated with the dataset?

**Comment by cchristelis** [comment]
For posterity: @ludwigschwardt and I had a Mattermost chat and decided that because the IntegrationTime is being used elsewhere we'll continue with this term.

**Pull request merged**